lib/plugins/aws/customResources/resources/apiGatewayCloudWatchRole/handler.js

```diff
@@ -68,40 +64,26 @@ async function create(event, context) {
     if (
       !attachedPolicies.some(policy => policy.PolicyArn === apiGatewayPushToCloudWatchLogsPolicyArn)
     ) {
-      await iam
-        .attachRolePolicy({
-          PolicyArn: apiGatewayPushToCloudWatchLogsPolicyArn,
-          RoleName: roleName,
-        })
-        .promise();
+      await awsRequest('IAM', 'attachRolePolicy', {
+        PolicyArn: apiGatewayPushToCloudWatchLogsPolicyArn,
+        RoleName: roleName,
+      });
     }
   }
 
   // there's nothing to do if the role is the same
   if (roleArn === assignedRoleArn) return null;
 
-  const updateAccount = async (counter = 1) => {
-    try {
-      await apiGateway
-        .updateAccount({
-          patchOperations: [
-            {
-              op: 'replace',
-              path: '/cloudwatchRoleArn',
-              value: roleArn,
-            },
-          ],
-        })
-        .promise();
-    } catch (error) {
-      if (counter < 10) {
-        // Observed fails with errors marked as non-retryable. Still they're outcome of
-        // temporary state where just created AWS role is not being ready for use (yet)
-        await wait(10000);
-        return updateAccount(++counter);
-      }
```

Comment:
I think we need to keep that try/catch, as in this case we want to catch also non-retryable errors, while `awsRequest` retries only if error is retryable